### PR TITLE
[#12271] Docs: Upgrade to latest MarkBind version

### LINKED ISSUE #12271: Docs: Upgrade to latest MarkBind version
MarkBind is the static-site generator currently used in the TM docs. The docs are on MarkBind v3.1.1; the latest MarkBind version is v4.1.0, with v5 on the horizon.

The new releases of MarkBind have several new feature additions and bug fixes - full release notes can be found [here](https://github.com/MarkBind/markbind/releases).

### PR DESCRIPTION
**Pull request by jingting1412**
Fixes #12271 

**Outline of Solution**
Upgraded the docs to the latest version of Markbind, v5.3.0 (full release notes can be found [here](https://github.com/MarkBind/markbind/releases)). 

I have deployed it from my branch [at this link](https://jingting1412.github.io/teammates/)

UI changes:
- Added breadcrumbs to the pages for users to see the hierarchy of the site
Screenshot 2024-03-13 at 1 50 47 PM

Other changes:
- Made navbar `sticky` so that it remains at the top of the page while scrolling. From testing on my deployed branch, using `<header fixed>` would sometimes not cause the header to remain at the top while scrolling, shown in the screenshot below which is at [this location on the page](https://jingting1412.github.io/teammates/development.html#managing-the-dev-server-back-end):
Screenshot 2024-03-13 at 4 58 54 PM

As the screenshot shows, the header is not present when I've scrolled down the page. As such I've changed it to `<header sticky>` to ensure that it would always stick to the top while scrolling

- Changed how `style.puml` is included in the puml diagrams. From testing on my own deployed branch, the include statement `!include diagrams/style.puml` would sometimes cause the path to be resolved wrongly [as shown in this commit](https://github.com/jingting1412/teammates/commit/4f40bb1361a91831d9b9bdfa11d2b13933ec02f6#diff-52426234bb64871f327069101a09f5a2991045dd4c7cd601f1ef76d4ce85d41e). When I changed the statement to just `!include style.puml`, the diagrams are correctly shown [as seen in this commit](https://github.com/jingting1412/teammates/commit/20a1e57ff641771fd337e9e1a7df0daa33d97fb7#diff-6e368c47a8a89daa48ed73150bd5dc9f6f158f7b6f775efaf0521c9e40d1315c)

### CODE REVIEW

**Comment by kaixin-hc** [comment]
Thanks for working on this @jingting1412 ! Would you revise the issue description so it is easier for teammates people to review? I think screenshots/video will help :)

Comparing the new appearance against the old, I think that the sticky header and working PUML are [actually already in the live dev docs](https://teammates.github.io/teammates/) - so **the only "change" in the UI is the breadcrumbs** (as auto-breadcrumbs being a MarkBind feature added in 5.0). Is that correct? (if it is, maybe can separate out the UI change from your explanations of why the other code changed 😅 

I understand that the `variables` file is not used on the teammates website and `variables.json` does not exist. Perhaps we should remove this file? (TEAMMATES people, if this is considered an unrelated change we can also file an issue and open a separate PR - let us know!)

**Comment by jingting1412** [comment]
thanks @kaixin-hc for the suggestion! I've done more testing on [my deployed branch](https://jingting1412.github.io/teammates/) and have since updated the description to better reflect the nature of the changes and to add images. 

Regarding the `variables.md` file, I think its up to the TEAMMATES team if they still want to keep it or not. Its a useful tool to have in case they would like to declare variables in the future. You can read more about it in our user guide [here](https://markbind.org/userGuide/reusingContents.html#global-variables)

**Review by cedricongjh — APPROVED** [review]
LGTM, thanks for providing a deployed version of the docs for us to look at!

**Review by ziqing26 — APPROVED** [review]
LGTM, thanks for the change! It is good to keep the variable file in case we need to declare some in the future.

File: package.json [hunk 1]
```diff
@@ -75,6 +75,7 @@
     "eslint-plugin-rxjs": "^5.0.3",
     "jest": "^29.7.0",
     "lintspaces-cli": "^0.7.1",
+    "markbind-cli": "^5.3.0",
```
Comment (wkurniawan07) [inline package.json:78]: This shouldn't be needed. Only the `package.json` file in `docs/` folder needs the change.

Comment (jingting1412) [reply package.json:78]: Hi can I clarify the reason? Isn’t it better for all package.json files to have the same latest version of markbind? 

On the same note, can I also clarify if I changed back this package.json file, I’ll also need to change back the package-lock.json?

Comment (wkurniawan07) [reply package.json:78]: @jingting1412 I believe you have figured it out, but the main project (thus root `package.json`) has nothing to do with MarkBind. This is not a Lerna monorepo project whereby the root `package.json` affects the one in the docs folder.

Comment (jingting1412) [reply package.json]: Yes understood thank you, I have since reverted the changes I've made to the main json files.

**Review by wkurniawan07 — APPROVED** [review]

**Pull request merged**